### PR TITLE
Bump Citus version into 13.0.0

### PR DESCRIPTION
**Pull request by naisila**
We are using `release-13.0` branch for both development and release, to deliver PG17 support in Citus.

Afterwards, we will (probably) merge this branch into main.

Some potential changes for main branch, after we are done working on release-13.0:
- Merge changes from `release-13.0` to `main`
- Figure out what changes were there on 12.2, move them to 13.1 version. In a nutshell: rename `12.1--12.2` to `13.0--13.1` and fix issues.
- Set version to 13.1devel

### CODE REVIEW

**Comment by hanefi** [comment]
Some initial high level comments:

- We usually use `X.Y-devel` on main branch, and `X.Y.Z` on release branches.
- I would expect to see the changes on the main branch first, and then a followup PR that changes the release branch. This is not set in stone, and can be changed if there is a new requirement that I do not yet know.
- I would expect to see the latest minor versions if we are updating supported PG versions in CI. i.e. 17.2.
- It might make sense to update the minor versions of other PG versions as well. Optionally on a separate PR.

File: src/backend/distributed/sql/citus--12.1-1--13.0-1.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- citus--12.1-1--13.0-1.sql
```
Comment (hanefi) [inline src/backend/distributed/sql/citus--12.1-1--13.0-1.sql:1]: On main branch the current version is '12.2-1' as seen in https://github.com/citusdata/citus/blob/665d72a2f57bf2c94241ac9892306ac7c70d098a/src/backend/distributed/citus.control#L3C19-L3C27

Currently you do not have a valid upgrade path from main branch to this release branch.

IMO, we should remove all references to 12.2 as we are skipping that version. I suggest we rename 12.2 to 13.0 on the main branch. After that make sure that similar changes are also applied on this release branch.

I guess, rebasing this release branch is not an option. Instead we can copy all the relevant sql files from the main branch that are relevant for 13.0-1 to this branch.

File: src/test/regress/citus_tests/common.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def get_pg_major_version():
 OLDEST_SUPPORTED_CITUS_VERSION_MATRIX = {
     14: "10.2.0",
     15: "11.1.5",
-    16: "12.1devel",
+    16: "12.1.5",
```
Comment (hanefi) [inline src/test/regress/citus_tests/common.py:95]: Would you like to add an entry for PG 17 here?

```suggestion
    16: "12.1.5",
    17: "13.0.0",
```

File: src/test/regress/expected/multi_extension.out [hunk 3]
```diff
@@ -1430,12 +1430,30 @@ SELECT * FROM multi_extension.print_extension_changes();
                  | function citus_schema_move(regnamespace,text,integer,citus.shard_transfer_mode) void
 (5 rows)
 
+-- Test downgrade to 12.1-1 from 13.0-1
+ALTER EXTENSION citus UPDATE TO '13.0-1';
+ALTER EXTENSION citus UPDATE TO '12.1-1';
+ERROR:  extension "citus" has no update path from version "13.0-1" to version "12.1-1"
```
Comment (hanefi) [inline src/test/regress/expected/multi_extension.out:1436]: This error message indicates that something is wrong with our downgrade script.

**Review by hanefi — CHANGES_REQUESTED** [review]

File: src/test/regress/citus_tests/common.py (hunk 2 above)
Comment (naisila) [reply src/test/regress/citus_tests/common.py]: I will add that in the PR where we add support for PG17 in the configure script as well.

**Review by hanefi — APPROVED** [review]

**Pull request merged**